Dockerfile

```diff
@@ -14,11 +14,13 @@ ENV WEB_SERVER_UWSGI_ASYNC_CORES 10
 ENV HTTP_KEEP_ALIVE 2
 ENV GUNICORN_CMD_ARGS -c gunicorn_config.py
 
-COPY Pipfile Pipfile
-COPY Pipfile.lock Pipfile.lock
+COPY pyproject.toml pyproject.toml
+COPY poetry.lock poetry.lock 
 
 RUN groupadd -r appuser && useradd -r -g appuser -u 9000 appuser && chown -R appuser:appuser .
-RUN pip install pipenv==2023.8.22 && pipenv install --deploy --system && make build
+RUN pip install "poetry==1.7.1"
+RUN poetry config virtualenvs.create false
+RUN poetry install --no-dev && make build
```

Comment:
Use `--only main` instead of `--no-dev` as poetry no allows groups. If we have more than a dev group, i.e test group, this would install test group dependencies. Plus `--no-dev` is deprecated.